### DISCUSSION TITLE
Can't debug in Codespaces when in the browser

### DISCUSSION THREAD

**Discussion by noelbundick** [community/community#5590] [Codespaces] [answered]
👋  Hello!

I have a public repo that is correctly configured with `launch.json` entries to debug a Rust app ([noelbundick-msft/hack-mcp-dx](https://github.com/noelbundick-msft/hack-mcp-dx))

When I launch Codespaces in the browser (or via app installed via browser address bar), I can usually hit `F5` and run my app successfully exactly one time. On the 2nd+ attempts, I *always* get an error dialog with the message: `Cannot start debugging because no launch configuration has been provided.`

![image](https://user-images.githubusercontent.com/1970162/132918084-adbbd7f4-2242-4c35-bbdc-fa1abe3331d9.png)

When I open `launch.json` to look at the configuration that worked once, then hit `F5`, VS Code gives me a different unexpected dialog that reads: `You don't have an extension for debugging 'JSON with Comments'. Should we find a 'JSON with Comments' extension in the Marketplace?`

![image](https://user-images.githubusercontent.com/1970162/132918144-ee048f7c-0766-4e37-9127-052743647f1e.png)

When I open my Codespace locally in VS Code, my `launch.json` config works correctly every time, and I can debug my app with no issues

Is this a bug / known issue? 

Thanks!

Noel

**Comment by bartr** [comment]
I'm having a similar issue with a simple C# project - wonder if it's the same issue?

**Comment by mattklup** [comment]
Same issue

**Comment by Chuxel** [comment]
@weinand Is this a known issue with the current stable release on web?

Version info (Codespaces Web):
------------
Version: 1.60.0
Commit: e7d7e9a9348e6a8cc8c03f877d39cb72e5dfb1ff
Date: 2021-09-01T10:49:24.188Z
Browser: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/93.0.4577.63 Safari/537.36 Edg/93.0.961.38

**Reply by roblourens** [reply]
I can repro this and am investigating

**Reply by weinand** [reply]
Any findings?

**Comment by richardkazuomiller** [comment]
Same issue here. I hope it can be fixed, because my use case specifically depends on running the debugger. For now, I'll need to advise my colleagues to open the project in VSCode locally.

**Comment by richardkazuomiller** [comment]
I was able to find a workaround for this by installing the [Commands](https://marketplace.visualstudio.com/items?itemName=fabiospampinato.vscode-commands) extension and setting up a custom command to launch my debugger like this:

```json
{
  "commands": [
    {
      "command": "commands.refresh",
      "text": "$(sync)",
      "tooltip": "Refresh commands",
      "color": "#FFCC00"
    },
    {
      "command": "debug.startFromConfig",
      "text": "Debug my thing",
      "arguments": [
        {
          "name": "Debug my thing",
          "args": [
            "${workspaceFolder}/scripts/my/script.js"
          ],
          "request": "launch",
          "cwd": "${workspaceRoot}",
          "type": "node",
          "protocol": "inspector",
          "internalConsoleOptions": "openOnSessionStart",
          "outputCapture": "std"
        }
      ]
    }
  ]
}

**Comment by Chuxel** [comment, marked as answer]
To close the loop here - A fix here was released in the VS Code stable release yesterday which will be landing in Codespaces soon.  In the mean time, you can use VS Code insiders which already has the fix.  To do so, click the gear in the lower left corner, and select "Switch to Insiders version..."

![image](https://user-images.githubusercontent.com/312252/136566788-253ed238-27f4-4ed7-bc59-901f3a4c080c.png)
